ls1intum/Artemis
# `Programming exercises`: Fix duplicated participant score update

### Checklist
#### General
#### Server

### Motivation and Context
We recently noticed that when processing new build results, the participant scores get updated twice. 

### Description
This is because the score update is bound to every result update database operation. When receiving the result, the database gets updated twice to link the result with its submission.

We now temporarily remove the participation in the first update. This will not trigger the first score update since participant scores are only scheduled for student participation.

### Steps for Testing
Prerequisites:
- 1 Student
- 1 Programming Exercise

1. Submit a new solution for a programming exercise
2. Verify in the logs that the participant scores were only updated once.
3. Check that the result is displayed correctly and linked to its participation.

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test9)

### Review Progress

#### Performance Review
#### Code Review
#### Manual Tests

### Test Coverage
new test added, coverage unchanged.
